### PR TITLE
Add bincrafters-templates script

### PR DESCRIPTION
**Pull request by madebr**
As a step between the current situation and the cookiecutter,
this pr adds a `bincrafters-templates` script

- tests are included
- I've adapted it from bincrafters-conventions
- a package is generated and is ready to be added to pypi (I propose bincrafters-templates)
- travis: https://travis-ci.com/madebr/conan-templates/builds/113686109
- appveyor: https://ci.appveyor.com/project/madebr/conan-templates/builds/24911576

Usage: (after install
`bincrafters-templates`: will generate a default template in the current directory
`bincrafters-templates --default`: same
`bincrafters-templates --header_only`: creates a header_only template in the current directory
`bincrafters-templates --installer_only`: creates an installer_only template in the current directory
`bincrafters-templates --help`: get help

/cc @Croydon @uilianries @SSE4

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      tags: true
+      condition: "$TRAVIS_OS_NAME = 'linux'"
+    skip_cleanup: true
+    skip_existing: true
+  - provider: pypi
+    user: ${TEST_PYPI_USERNAME}
+    server: https://test.pypi.org/legacy/
+    password: ${TEST_PYPI_PASSWORD}
+    on:
+      branch: master
```
Comment (uilianries) [inline .travis.yml]: release is better

**Comment by uilianries** [comment]
Not sure if we should create a new project called bincrafters-templates

File: .ci/install.sh [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-set -ex
+set -e
+set -x
 
 if [[ "$(uname -s)" == 'Darwin' ]]; then
     brew update || brew update
     brew outdated pyenv || brew upgrade pyenv
     brew install pyenv-virtualenv
-    brew install cmake || brew upgrade cmake || true
```
Comment (SSE4) [inline .ci/install.sh]: why skip cmake?

**Comment by Croydon** [comment]
I'm still unsure if I'm supporting a new template generator script, but **IF** then I guess I would prefer to keep it in another repository, to keep this one really clean and simple

**Comment by SSE4** [comment]
let's create new repository then

File: .travis.yml (hunk 1 above)
Comment (madebr) [reply .travis.yml]: This is a copy of `.travis.yml` of `bincrafters-conventions`:
https://github.com/bincrafters/bincrafters-conventions/blob/17107ace2dd55a368a7df198325c4d52176a1d75/.travis.yml#L34-L37

File: .ci/install.sh (hunk 2 above)
Comment (madebr) [reply .ci/install.sh]: Copy of `.ci/install.sh` of `conan-templates`:
https://github.com/bincrafters/conan-templates/blob/387bf5d7a98fe5810c0438a502d8ba528f92d5b2/.ci/install.sh#L9

**Comment by Croydon** [comment]
Conan 0.16.0 has extended its own template feature. Maybe we can build upon that somehow?

**Comment by madebr** [comment]
@Croydon  https://docs.conan.io/en/latest/reference/commands/creator/new.html?highlight=template#
It looks like it only supports to use a specific templated `conanfile.py`. No special pre- or posthooks.

**Comment by uilianries** [comment]
https://github.blog/2019-06-06-generate-new-repositories-with-repository-templates/

I think GitHub solved our problem.

**Comment by Croydon** [comment]
> https://github.blog/2019-06-06-generate-new-repositories-with-repository-templates/
I think GitHub solved our problem.

I just wrote an email to them, suggesting more custom rules for this template generator. Right now we would need to maintain each template in an own repository which makes it a lot harder to discover and maintain....

**Comment by uilianries** [comment]
> > [github.blog/2019-06-06-generate-new-repositories-with-repository-templates](https://github.blog/2019-06-06-generate-new-repositories-with-repository-templates/)
> > I think GitHub solved our problem.
> 
> I just wrote an email to them, suggesting more custom rules for this template generator. Right now we would need to maintain each template in an own repository which makes it a lot harder to discover and maintain....

Nice! Thanks for doing this.

**Comment by Croydon** [comment]
@madebr Could we make this with reasonable effort in a way, that 
a) we move the script to another repository
b) and write the script in a way that it directly uses the templates files from this directory, so that we don't have redundant information?

**Comment by madebr** [comment]
@Croydon 
I have something at https://github.com/madebr/conan-templates-script
travis-ci at https://travis-ci.com/madebr/conan-templates-script/builds/114877662

**Comment by SSE4** [comment]
see https://github.com/bincrafters/community/issues/826

**Pull request closed without merging**